pkg/virt-handler/vm.go

```diff
@@ -87,66 +87,29 @@ func (d *VMHandlerDispatch) Execute(store cache.Store, queue workqueue.RateLimit
 			return
 		}
 		vm = v1.NewVMReferenceFromName(name)
+	} else {
+		vm = obj.(*v1.VM)
+	}
 
+	// Check For Migration before processing vm not in our cache
+	if !exists {
 		// If we don't have the VM in the cache, it could be that it is currently migrating to us
-		result := d.restClient.Get().Name(vm.GetObjectMeta().GetName()).Resource("vms").Namespace(kubeapi.NamespaceDefault).Do()
-		if result.Error() == nil {
-			// So the VM still seems to exist
-			fetchedVM, err := result.Get()
-			if err != nil {
-				// Since there was no fetch error, this should have worked, let's back off
-				queue.AddRateLimited(key)
-				return
-			}
-			if fetchedVM.(*v1.VM).Status.MigrationNodeName == d.host {
-				// OK, this VM is migrating to us, don't interrupt it
-				queue.Forget(key)
-				return
-			}
-		} else if result.Error().(*errors.StatusError).Status().Code != int32(http.StatusNotFound) {
-			// Something went wrong, let's try again later
+		isDestination, err := d.isMigrationDestination(vm.GetObjectMeta().GetName())
+		if err != nil {
+			// unable to determine migration status, we'll try again later.
 			queue.AddRateLimited(key)
 			return
+		} else if isDestination {
```

Comment:
The error handling block returns so this doesn't need to be an "else" block. I'd find the code easier to follow without it, but that's simply personal preference.